src/components/header/AppHeader.vue

```diff
@@ -0,0 +1,536 @@
+        ...mapMutations('login', [
+            'setLogin',
+        ]),
+        scrollHandler(info) {
+            this.menuHiddenDesktop = info.direction === 'down';
+        },
+        goTo(to) {
+            this.mobileMenuIsOpen = false;
+            this.advancedMenuExpanded = false;
+
+            const httpsRegex = /^https/;
+            if (typeof to === 'string' && httpsRegex.test(to)) {
+                window.location.href = to;
+                return;
+            }
+
+            this.$router.push(to);
+        },
+        handleClickaway() {
+            this.mobileMenuIsOpen = false;
+            this.advancedMenuExpanded = false;
+        },
+        toggleDarkMode() {
+            this.$q.dark.toggle();
+            localStorage.setItem('darkModeEnabled', this.$q.dark.isActive);
+        },
+        handleLoginLogout() {
+            if (this.isLoggedIn) {
+                this.logout();
+            } else {
+                this.showLoginModal = true;
+            }
+        },
+        logout() {
+            if (this.isNative) {
+                const loginData = localStorage.getItem('loginData');
+                if (!loginData)
+                    return;
+
+                const loginObj = JSON.parse(loginData);
+                const wallet = this.$ual.authenticators.find(a => a.getName() == loginObj.provider);
+                wallet.logout();
+            }
+
+            this.setLogin({});
+            localStorage.removeItem('loginData');
+            this.$providerManager.setProvider(null);
+        },
+    },
+}
+</script>
+
+<style lang="scss">
+.c-header {
+    $this: &;
+
+    --background-color: white;
+    --text-color: #{$dark};
+    --highlight-color: #{$grey-3};
+
+    background-color: var(--background-color);
+    position: fixed;
+    top: 0;
+    right: 0;
+    left: 0;
+    height: 48px;
+    padding: 0 0 0 16px;
+    z-index: 999;
+    color: var(--text-color);
+
+    display: flex;
+    flex-wrap: nowrap;
+    justify-content: space-between;
+
+    @media screen and (min-width: $breakpoint-lg-min) {
+        height: 64px;
+        padding: 0 24px;
+    }
+
+    @at-root .body--dark & {
+        --background-color: #{$dark};
+        --text-color: white;
+        --highlight-color: #{$grey-9};
+    }
+
+    &__logo-container {
+        width: 48px;
+        height: 48px;
+
+        @media screen and (min-width: $breakpoint-lg-min) {
+            height: 64px;
+            gap: 12px;
+            color: var(--text-color);
+            font-size: 18px;
+        }
+    }
+
+    &__logo-image-container {
+        position: relative;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+    }
+
+    &__testnet-indicator {
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        margin: auto;
+        font-size: 10px;
+        height: min-content;
+        background: rgba($dark, 0.4);
+        color: white;
+    }
+
+    &--logo-text {
```

Comment:
Same thing about the BEM sintaxis

```suggestion
    &__logo-text {
```